internal/internal_decision_state_machine.go

```diff
@@ -788,6 +819,37 @@ func (h *decisionsHelper) handleRequestCancelExternalWorkflowExecutionFailed(wor
 	}
 }
 
+func (h *decisionsHelper) signalExternalWorkflowExecution(domain, workflowID, runID, signalName string, input []byte, signalID string) decisionStateMachine {
+	attributes := &s.SignalExternalWorkflowExecutionDecisionAttributes{
+		Domain:     common.StringPtr(domain),
+		WorkflowId: common.StringPtr(workflowID),
+		RunId:      common.StringPtr(runID),
+		SignalName: common.StringPtr(signalName),
+		Input:      input,
+		Control:    []byte(signalID),
+	}
+	decision := newSignalExternalWorkflowStateMachine(attributes, signalID)
+	h.addDecision(decision)
+	return decision
+}
+
+func (h *decisionsHelper) handleSignalExternalWorkflowExecutionInitiated(signalID string) {
+	decision := h.getDecision(makeDecisionID(decisionTypeSignal, signalID))
+	decision.handleInitiatedEvent()
+}
+
+func (h *decisionsHelper) handleSignalExternalWorkflowExecutionRequested(signalID string) decisionStateMachine {
```

Comment:
rename to handleSignalExternalWorkflowExecutionCompleted

Reply:
Renamed